### PR TITLE
debounce send step

### PR DESCRIPTION
**Pull request by schehata**
Here is my thinking on how could this be done, but I am still trying to figure out first how to achieve those steps:

1. collect the metrics when ever report method is called (some state?)
2. start the timer when report is called, send the collected metrics and clear the timer
3. what if the report method is called while the the requests are being sent, or the collected metrics are being cleared?

### CODE REVIEW

**Comment by schehata** [comment]
@bahlo any ideas to improve that test? I wasn't able to fully understand how to test with the `Spy`.

**Review by bahlo — COMMENTED** [review]
Debounce logic looks good 🙌

File: __tests__/debounce.test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        reportWebVitals(metric)
+    }
+
+    await new Promise(res => 
+    setTimeout(() => {
+        const spy = jest.spyOn(global, 'fetch')
+        expect(spy).toHaveBeenCalledTimes(1)
+        res(true)
+    }, 4000))
+})
```
[the remark this answers is not in the document]
Comment (schehata) [reply __tests__/debounce.test.ts]: done

File: index.ts [hunk 2]
```diff
@@ -31,18 +34,31 @@ export function withAxiomProxy(nextConfig: NextConfig): NextConfig {
   };
 }
 
+export function getCurrentPage() {
+  return window.__NEXT_DATA__?.page;
+}
```
Comment (bahlo) [inline index.ts:39]: Do we still need this function?

File: __tests__/debounce.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    [
+      { id: '1', startTime: 1234, value: 1, name: 'FCP', label: 'web-vital' },
+      { id: '2', startTime: 5678, value: 2, name: 'FCP', label: 'web-vital' },
+    ],
+    [{ id: '3', startTime: 9012, value: 3, name: 'FCP', label: 'web-vital' }],
+    [{ id: '4', startTime: 4012, value: 4, name: 'FCP', label: 'web-vital' }],
+  ];
+
+  metricsMatrix[0].forEach(reportWebVitals);
+  expect(fetch).toBeCalledTimes(0);
```
Comment (bahlo) [inline __tests__/debounce.test.ts:23]: Can we move this right before line 29?

**Review by bahlo — APPROVED** [review]
two small things, then we can land this 💯

**Comment by bahlo** [comment]
Oh and can you add the following to `.npmignore`?
```gitignore
/__tests__
/jest.config.ts
/tsconfig.json
```

This is how it looks when installed using npm:
Screenshot 2022-05-24 at 08 58 14

File: index.ts (hunk 2 above)
Comment (schehata) [reply index.ts]: removed

File: __tests__/debounce.test.ts (hunk 3 above)
Comment (schehata) [reply __tests__/debounce.test.ts]: done

**Pull request merged**